contracts/vaults/RibbonThetaVault.sol

```diff
@@ -321,76 +318,128 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         VaultDeposit.DepositReceipt memory depositReceipt =
             depositReceipts[msg.sender];
 
+        // If we have an unprocessed pending deposit from the previous rounds, we have to process it.
+        uint128 unredeemedShares = depositReceipt.unredeemedShares;
+        if (
+            depositReceipt.round > 0 &&
+            depositReceipt.round < currentRound &&
+            !depositReceipt.processed
+        ) {
+            unredeemedShares = _getSharesFromReceipt(depositReceipt);
+        }
+
         // If we have a pending deposit in the current round, we add on to the pending deposit
         if (currentRound == depositReceipt.round) {
             // No deposits allowed until the next round
             require(!depositReceipt.processed, "Processed");
 
             uint256 newAmount = uint256(depositReceipt.amount).add(amount);
-            require(newAmount < type(uint128).max, "Overflow");
+            require(newAmount < type(uint104).max, "Overflow");
 
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
                 round: currentRound,
-                amount: uint128(newAmount)
+                amount: uint104(newAmount),
+                unredeemedShares: unredeemedShares
             });
         } else {
-            require(amount < type(uint128).max, "Overflow");
+            require(amount < type(uint104).max, "Overflow");
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
                 round: currentRound,
-                amount: uint128(amount)
+                amount: uint104(amount),
+                unredeemedShares: unredeemedShares
             });
         }
 
         _totalPending = _totalPending.add(amount);
+    }
 
-        // If we have an unprocessed pending deposit from the previous rounds, we have to redeem it.
-        if (
-            depositReceipt.round > 0 &&
-            depositReceipt.round < currentRound &&
-            !depositReceipt.processed
-        ) {
-            _redeemDeposit(currentRound, depositReceipt, false);
-        }
+    /**
+     * @notice Redeems shares that are owed to the account
+     * @param shares is the number of shares to redeem
+     */
+    function redeem(uint256 shares) external nonReentrant {
+        require(shares > 0, "!shares");
+        _redeem(shares, false);
     }
 
-    function redeemDeposit() external nonReentrant {
+    /**
+     * @notice Redeems the entire unredeemedShares balance that is owed to the account
+     */
+    function maxRedeem() external nonReentrant {
```

Comment:
maybe just have a redeem() and in the frontend we have max button + total unredeemed calculation off-chain?

Reply:
I think the redemption functionality is mostly useful for other contracts. For users that use the FE, they don't have to call the `redeem` function in the first place.

If we're writing this function to cater towards contracts, they would appreciate a `maxRedeem` function because it's more gas efficient